riscv-runtime/src/std_support.rs

```diff
@@ -14,30 +14,36 @@ use powdr_riscv_syscalls::Syscall;
 use crate::io::write_slice;
 use getrandom::{register_custom_getrandom, Error};
 
+/// The std interface to random number generation.
 #[no_mangle]
 extern "C" fn sys_rand(buf: *mut u32, words: usize) {
-    // This is only used by std to key the hash function of HashMap. Until we
-    // figure a way to make random values available to user programs, let's just
-    // have deterministic HashMaps.
-    const VALUE: u32 = 4;
-
-    let slice = unsafe { slice::from_raw_parts_mut(buf, words) };
-    for v in slice.iter_mut() {
-        *v = VALUE;
+    let buf = buf as *mut u8;
+    unsafe {
+        let slice = slice::from_raw_parts_mut(buf, words * 4);
+        powdr_getrandom_impl(slice);
     }
 }
 
-fn powdr_getrandom(s: &mut [u8]) -> Result<(), Error> {
-    // TODO: getrandom expects this u8 slice as input, so we're not using sys_rand.
-    // Both should probably be fixed/unified later once we have a proper RNG.
-    const VALUE: u8 = 3;
-    s.iter_mut().for_each(|v| *v = VALUE);
-
+/// The de-factor standard rust interface to low level random number generation.
+fn powdr_getrandom(buf: &mut [u8]) -> Result<(), Error> {
+    powdr_getrandom_impl(buf);
     Ok(())
 }
-
 register_custom_getrandom!(powdr_getrandom);
 
+/// This is a placeholder to pretend to provide a random number generator, for
+/// places like the hash function of HashMap who needs something.
+///
+/// This could be improved by using the rand_chacha crate, but I think it is
+/// worse, as it will just mask the fact that we are not providing a real
+/// entropy source.
```

Comment:
does it make sense to borrow from risc0's approach, with the disclaimer + feature flag or panic otherwise?

Reply:
yes, let me do that